src/mmtk.rs

```diff
@@ -349,6 +433,8 @@ impl<VM: VMBinding> MMTK<VM> {
         self.state
             .internal_triggered_collection
             .store(true, Ordering::Relaxed);
+        // TODO: The current `GCRequester::request()` is probably incorrect for internally triggered GC.
```

Comment:
With this PR applied, `GCRequester` is now used for *mutator* threads to communicate with *GC worker* threads (there is no coordinator).  It does so by setting a variable ("goal") in `WorkerMonitor` and notifying one GC worker.  For optimization, `GCRequester` also uses an `AtomicBool` to filter out consecutive requests from mutators.

But "internally triggered GC" is used to upgrade a concurrent GC to a STW GC.  That is not done by communication between mutators and GC threads, but by GC threads coordinating themselves.  They still communicate via shared data structures such as `WorkerMonitor`, but setting a "goal" is probably not the right way to perform the "upgrading" because GC threads only respond to one goal at a time.

I am still not sure how to do such "upgrading" in Rust MMTk.  The `lxr` branch is not using the `trigger_internal_collection_request` function, either, although `GCRequester::request(&self, concurrent: bool)` has an additional argument `concurrent: bool` in the `lxr` branch.